bouncer/src/repo/lib/repo_license.cpp

```diff
@@ -0,0 +1,169 @@
+/**
+*  Copyright (C) 2021 3D Repo Ltd
+*
+*  This program is free software: you can redistribute it and/or modify
+*  it under the terms of the GNU Affero General Public License as
+*  published by the Free Software Foundation, either version 3 of the
+*  License, or (at your option) any later version.
+*
+*  This program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+*  GNU Affero General Public License for more details.
+*
+*  You should have received a copy of the GNU Affero General Public License
+*  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "repo_license.h"
+
+namespace Licensing
+{
+	// definitions for class static variables
+	repo::lib::RepoUUID LicenseValidator::instanceId;
+	std::string LicenseValidator::licenseStr;
+	std::unique_ptr<Cryptolens> LicenseValidator::cryptolensHandle;
+
+	void LicenseValidator::RunActivation()
+	{
+#ifdef REPO_LICENSE_CHECK
+		// only run once 
+		if (!instanceId.isDefaultValue() && cryptolensHandle) return;
+
+		// Setting up the handle
+		licenseStr = GetLicenseString();
+		cryptolens::Error e;
+		cryptolensHandle = std::make_unique<Cryptolens>(e);
+		cryptolensHandle->signature_verifier.set_modulus_base64(e, pubKeyModulus);
+		cryptolensHandle->signature_verifier.set_exponent_base64(e, pubKeyExponent);
+		// Using machine code to store the instance UUID instead, as per
+		// https://help.cryptolens.io/licensing-models/containers
+		instanceId = repo::lib::RepoUUID::createUUID();
+		cryptolensHandle->machine_code_computer.set_machine_code(e, instanceId.toString());
+
+		// License activation 
+		cryptolens::optional<cryptolens::LicenseKey> licenseKey =
+			cryptolensHandle->activate_floating
+			(
+				e, // Object used for reporting if an error occured
+				authToken, // Cryptolens Access Token
+				productId, // Product key associated with the 3drepo.io pipeline
+				licenseStr, // License Key
+				floatingTimeIntervalSec // The amount of time the user has to wait before an actived machine (or session in our case) is taken off the license.
+			);
+
+		// dealing with early bail out scenarios
+		repoInfo << "****License activation summary****";
```

Comment:
```suggestion
		repoInfo << "****License activation summary****";
```

If we're adding a header block like this then we should add an end of block comment at the end of the function before we return.